packages/@react-spectrum/slider/src/SliderBase.tsx

```diff
@@ -0,0 +1,197 @@
+  if (alwaysDisplaySign) {
+    if (formatOptions != null) {
+      if (!('signDisplay' in formatOptions)) {
+        // @ts-ignore
+        formatOptions.signDisplay = 'exceptZero';
+      }
+    } else {
+      // @ts-ignore
+      formatOptions = {signDisplay: 'exceptZero'};
+    }
+  }
+
+  let state = useSliderState({...props, formatOptions});
+  let trackRef = useRef();
+  let {
+    containerProps,
+    trackProps,
+    labelProps
+  } = useSlider({...props, direction}, state, trackRef);
+
+  let inputRefs = [];
+  let thumbProps = [];
+  let inputProps = [];
+  for (let i = 0; i < count; i++) {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    inputRefs[i] = useRef();
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    let v = useSliderThumb({
+      index: i,
+      isDisabled: props.isDisabled,
+      trackRef,
+      inputRef: inputRefs[i],
+      direction
+    }, state);
+
+    inputProps[i] = v.inputProps;
+    thumbProps[i] = v.thumbProps;
+  }
+
+  let ticks = null;
+  if (tickCount > 0) {
+    let tickList = [];
+    for (let i = 0; i < tickCount; i++) {
+      let tickLabel = tickLabels ? tickLabels[i] : state.getFormattedValue(state.getPercentValue(i / (tickCount - 1)));
+      tickList.push(
+        <div className={classNames(styles, 'spectrum-Slider-tick')} key={i}>
+          {showTickLabels &&
+            <div className={classNames(styles, 'spectrum-Slider-tickLabel')}>
+              {tickLabel}
+            </div>
+          }
+        </div>
+      );
+    }
+    ticks = (<div className={classNames(styles, 'spectrum-Slider-ticks')}>
+      {tickList}
+    </div>);
+  }
+
+  let formatter = useNumberFormatter(formatOptions);
+
+  let displayValue = valueLabel;
+  let maxLabelLength = undefined;
+  if (!displayValue) {
+    maxLabelLength = Math.max([...formatter.format(state.getThumbMinValue(0))].length, [...formatter.format(state.getThumbMaxValue(0))].length);
+    switch (state.values.length) {
+      case 1:
+        displayValue = state.getThumbValueLabel(0);
+        break;
+      case 2:
+        // This should really use the NumberFormat#formatRange proposal...
+        // https://github.com/tc39/ecma402/issues/393
+        // https://github.com/tc39/proposal-intl-numberformat-v3#formatrange-ecma-402-393
+        displayValue = `${state.getThumbValueLabel(0)} - ${state.getThumbValueLabel(1)}`;
+
+        maxLabelLength = 2 + 2 * Math.max(
+          maxLabelLength,
+          [...formatter.format(state.getThumbMinValue(1))].length, [...formatter.format(state.getThumbMaxValue(1))].length
+        );
+        break;
+      default:
+        throw new Error('Only sliders with 1 or 2 handles are supported!');
+    }
+  }
+
+  let labelNode = <label className={classNames(styles, 'spectrum-Slider-label')} {...labelProps}>{props.label}</label>;
+  let valueNode = (<div
+    className={classNames(styles, 'spectrum-Slider-value')}
+    // TODO really?
+    role="textbox"
```

Comment:
@majornista do we need the slider value to be a textbox or should it be just a simple text element?

@mischnic if textbox, then we should probably add this to the `useSlider` hook as e.g. `valueProps` so others outside react-spectrum can make use of it...

Reply:
I went with the `textbox` so that it could be labeled by the label for the slider. Perhaps a better element, since it is read-only would be `output`, but `output` has some accessibility concerns, because [some screen-readers treat it as a live region](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/output#Accessibility_Concerns)?

Reply:
Ok. Going to merge this PR as is. Created #1146 to track moving this into the `useSlider` hook.